### ISSUE TITLE
libprojectm needs update, current version will break in std=c++17 builds

### ISSUE THREAD

**Issue by marcusmueller** [clementine-player/Clementine#7292]
### Before posting

Please follow the steps below and check the boxes with [x] once you did the step.

- [x] I checked the issue tracker for similar issues
- [x] I checked the [changelog](https://github.com/clementine-player/Clementine/blob/master/Changelog) if the issue is already resolved
- [x] I tried the latest Clementine build from [here](https://builds.clementine-player.org/)

### System information

Please provide information about your system and the version of Clementine used.

- Operating System: N/A
- Clementine version: N/A

### Expected behaviour / actual behaviour

It's 2023; bundled dependencies should not break with C++17 :)

### Steps to reproduce the problem (only for bugs)

running a test build of clementine with the usual warnings enabled (`cmake -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -DCMAKE_BUILD_TYPE=RelWithDebInfo -GNinja -DCMAKE_CXX_COMPILER_LAUNCHER=ccache -DCMAKE_CXX_COMPILER=clang++`), I spot that the currently bundled version of libprojectm still uses `std::binary_function`, which was [removed in C++17](https://en.cppreference.com/w/cpp/utility/functional/binary_function). Upstream master does no longer do that.

Also, there's a bunch of other very questionable things in that bundled versions, like an inherently broken random number generator, a lot of inherently false comparisons (a bool is never `< 0`), and so on. 

Question is why we're actually bundling libprojectm; don't think that makes too much sense. Same, but much more so, for QSQLite, which should probably just be a dependency.

**Comment by polsvoice** [comment]
Isn't libprojectm needed for visualizations?

**Comment by marcusmueller** [comment]
@polsvoice yes? So, the question is why we're *bundling* it, not why we're *using* it.

**Comment by DarthGandalf** [comment]
Yeah, visualizations work fine with libprojectm installed via the distro package manager
